Purchases/Public/RCPurchases.m

```diff
@@ -732,9 +746,9 @@ - (void)offeringsWithCompletionBlock:(RCReceiveOfferingsBlock)completion
     if (self.deviceCache.cachedOfferings) {
         RCDebugLog(@"Vending offerings from cache");
         CALL_IF_SET_ON_MAIN_THREAD(completion, self.deviceCache.cachedOfferings, nil);
-        if ([self.deviceCache isCacheStale]) {
-            RCDebugLog(@"Cache is stale, updating caches");
-            [self updateCaches];
+        if (self.deviceCache.isOfferingsCacheStale) {
+            RCDebugLog(@"Offerings cache is stale, updating cache");
+            [self updateOfferingsCache:completion];
```

Comment:
The completion block is going to get called twice. First time with the cached offerigns and then when the caches finish updating. This should be `[self updateOfferingsCache:nil];`